daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/PackedBinaryTraits.scala

```diff
@@ -95,24 +105,25 @@ abstract class PackedBinaryDecimalBaseParser(
     }
 
     try {
-      val bigDec = toBigDecimal(dis.getByteArray(nBits, start), binaryDecimalVirtualPoint)
-      start.simpleElement.overwriteDataValue(bigDec)
+      val dec = toPrimType(context, dis.getByteArray(nBits, start))
+      start.simpleElement.setDataValue(dec)
     } catch {
       case n: NumberFormatException => PE(start, "Error in packed data: \n%s", n.getMessage())
+      case i: InvalidPrimitiveDataException => PE(start, "Error in packed data: \n%s", i.getMessage())
     }
   }
 }
 
 abstract class PackedBinaryIntegerBaseParser(
   override val context: ElementRuntimeData
 ) extends PrimParser
-  with PackedBinaryConversion
+  with PackedBinaryConversion[JBigInteger]
   with PackedBinaryLengthCheck {
   override lazy val runtimeDependencies = Vector()
 
   val signed = {
```

Comment:
This val isn't used and can be deleted? Which I think makes sense because signedness is now checked by `primType.fromNumber`?